### DISCUSSION TITLE
VPN under BBN OS

### DISCUSSION THREAD

**Discussion by HawlAway** [bareboat-necessities/lysmarine_gen#453] [Q&A]
Is there any further information on the add VPN option which appears under the Network section of the Budgie control centre?  When it asks for a file is this a key file or something else?  

I have BBN OS installed on a Pi5 and I want to be able to VPN out to my home network from my boat.

**Comment by mgrouch** [comment]
You are free to choose any VPN software which runs on Debian. There are plenty of options

**Reply by HawlAway** [reply]
Thanks.  I tried installing OpenVPN on the pi and copied across a profile file from my VPN server also running OpenVPN.  I started OpenVPN service on the client Pi and rebooted.  However, I still cannot add a VPN connection.  The screen shot shows the the Budgie control centre screen for adding a VPN, but when I select a vpn conf file I get the error shown in the second screenshot.
![Screenshot_2024-09-09_12-20-41](https://github.com/user-attachments/assets/0bd9976e-ff4a-4a29-9e03-5683251e156a)

![Screenshot_2024-09-09_16-14-25](https://github.com/user-attachments/assets/f2d318b3-251f-4d6b-919c-6503839762f8)

I am not an expert, so any wisdom you might shed on the blockers would be approciated.

**Comment by mgrouch** [comment]
Instead of setting it in budgie control center try setting VPN in NetworkManager. 
NetworkManager is what is used for control WiFi

**Comment by HawlAway** [comment]
If by NetworkManager you mean the thing found in the upper right with the add VPN connection on the drop down menu, then this also does not work.  When trying to add a VPN using this tool, the drop down list of available VPNs is blank.Sent from my iPadOn 9 Sep 2024, at 17:22, Mikhail Grushinskiy ***@***.***> wrote:﻿
Instead of setting it in budgie control center try setting VPN in NetworkManager.
NetworkManager is what is used to control WiFi

**Reply by HawlAway** [reply]
Update following suggestion of using Network Manager.
After some more online research I established the need to install a network manager plug-in for openvpn.  After doing this I could now add the profile file and connect via VPN.  All looked OK but my Pi can't reach the internet through the VPN.  Some more investigation is needed, but progress has been made so thanks for he suggestions so far.
